crates/ruff_linter/src/rules/pylint/rules/nan_comparison.rs

```diff
@@ -0,0 +1,104 @@
+use ruff_python_ast::{self as ast, Expr};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for comparisons against NaN values.
+///
+/// ## Why is this bad?
+/// Comparing against a NaN value will always return False even if both values are NaN.
+///
+/// ## Example
+/// ```python
+/// if x == float("NaN"):
+///     pass
+/// ```
+///
+/// Use instead:
+/// ```python
+/// import math
+///
+/// if math.isnan(x):
+///     pass
+/// ```
+///
+#[violation]
+pub struct NanComparison {
+    using_numpy: bool,
+}
+
+impl Violation for NanComparison {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let NanComparison { using_numpy } = self;
+        if *using_numpy {
+            format!("Comparing against a NaN value, consider using `np.isnan`")
+        } else {
+            format!("Comparing against a NaN value, consider using `math.isnan`")
+        }
+    }
+}
+
+fn is_nan_float(expr: &Expr) -> bool {
```

Comment:
Also interested to hear if there is a better way to do this

Reply:
I tweaked it a little bit but it's generally correct. Kind of tedious to write out, but correct!